core/src/main/java/com/scalar/db/common/AbstractDistributedTransactionManager.java

```diff
@@ -183,6 +184,18 @@ public void rollback() throws RollbackException {
       }
     }
 
+    @Override
+    public void abort() throws AbortException {
+      if (status == Status.COMMITTED || status == Status.ROLLED_BACK) {
+        throw new IllegalStateException("The transaction has already been committed or aborted");
```

Comment:
Just noticed users would want to know the actual state in the log message?

Reply:
That's correct. Since we need to improve the messages in other parts, let's do that in another PR. Thank for pointing it out!